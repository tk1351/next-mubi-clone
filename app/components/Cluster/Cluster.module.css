.cluster {
    display: flex;
    flex-wrap: wrap;
    gap: 1rem;
}

.cluster-justify-start {
    justify-content: flex-start;
}

.cluster-aline-start {
    align-items: flex-start;
}

.cluster-justify-end {
    justify-content: flex-end;
}

.cluster-aline-end {
    align-items: flex-end;
}

.cluster-justify-center {
    justify-content: center;
}

.cluster-aline-center {
    align-items: center;
}
