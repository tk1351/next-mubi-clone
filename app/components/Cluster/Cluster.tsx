import { type FC, type PropsWithChildren } from "react";
import clsx from "clsx";
import styles from "./Cluster.module.css";

type ClusterPosition = "flex-start" | "flex-end" | "center";

export type Props = PropsWithChildren<{
  justify?: ClusterPosition;
  aline?: ClusterPosition;
}>;

export const Cluster: FC<Props> = ({
  children,
  justify = "flex-start",
  aline = "flex-start",
}) => {
  const clusterClassName = clsx(styles.cluster, {
    [styles["cluster-justify-start"]]: justify === "flex-start",
    [styles["cluster-justify-end"]]: justify === "flex-end",
    [styles["cluster-justify-center"]]: justify === "center",
    [styles["cluster-aline-start"]]: aline === "flex-start",
    [styles["cluster-aline-end"]]: aline === "flex-end",
    [styles["cluster-aline-center"]]: aline === "center",
  });
  return <ul className={clusterClassName}>{children}</ul>;
};
