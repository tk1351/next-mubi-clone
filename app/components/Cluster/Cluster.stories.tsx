import type { Meta, StoryObj } from "@storybook/react";
import { Cluster } from "./Cluster";

const meta = {
  title: "Components/Cluster",
  component: Cluster,
} satisfies Meta<typeof Cluster>;

export default meta;
type Story = StoryObj<typeof meta>;
export const FlexStart: Story = {
  args: {
    children: (
      <>
        <li>Fugiat veniam sunt ea incididunt.</li>
        <li>Qui in dolore incididunt sunt adipisicing ullamco.</li>
        <li>Elit duis cupidatat amet eiusmod ullamco.</li>
      </>
    ),
  },
};

export const FlexEnd: Story = {
  args: {
    children: (
      <>
        <li>Fugiat veniam sunt ea incididunt.</li>
        <li>Qui in dolore incididunt sunt adipisicing ullamco.</li>
        <li>Elit duis cupidatat amet eiusmod ullamco.</li>
      </>
    ),
    position: "right",
  },
};

export const Center: Story = {
  args: {
    children: (
      <>
        <li>Fugiat veniam sunt ea incididunt.</li>
        <li>Qui in dolore incididunt sunt adipisicing ullamco.</li>
        <li>Elit duis cupidatat amet eiusmod ullamco.</li>
      </>
    ),
    position: "center",
  },
};
