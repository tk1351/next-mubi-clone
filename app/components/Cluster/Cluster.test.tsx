import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { Cluster } from "./Cluster";

describe(`Testing Cluster Component`, () => {
  describe("Snapshot", () => {
    it("justify, aline が 'flex-start' の場合", () => {
      const { container } = render(
        <Cluster justify="flex-start" aline="flex-start">
          Cluster
        </Cluster>,
      );
      expect(container).toMatchSnapshot();
    });
    it("justify, aline が 'flex-end' の場合", () => {
      const { container } = render(
        <Cluster justify="flex-end" aline="flex-end">
          Cluster
        </Cluster>,
      );
      expect(container).toMatchSnapshot();
    });
    it("justify, aline が center の場合", () => {
      const { container } = render(
        <Cluster justify="center" aline="center">
          Cluster
        </Cluster>,
      );
      expect(container).toMatchSnapshot();
    });
  });
  describe("a11y", () => {
    it('role="list" であること', () => {
      render(<Cluster>Cluster</Cluster>);
      expect(screen.getByRole("list")).toBeInTheDocument();
    });
  });
});
