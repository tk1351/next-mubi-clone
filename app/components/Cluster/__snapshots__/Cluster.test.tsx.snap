// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Cluster Component > Snapshot > justify, aline が 'flex-end' の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-justify-end_b5f122 _cluster-aline-end_b5f122"
  >
    Cluster
  </ul>
</div>
`;

exports[`Testing Cluster Component > Snapshot > justify, aline が 'flex-start' の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-justify-start_b5f122 _cluster-aline-start_b5f122"
  >
    Cluster
  </ul>
</div>
`;

exports[`Testing Cluster Component > Snapshot > justify, aline が center の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-justify-center_b5f122 _cluster-aline-center_b5f122"
  >
    Cluster
  </ul>
</div>
`;

exports[`Testing Cluster Component > Snapshot > position が center の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-center_b5f122"
  >
    Cluster
  </ul>
</div>
`;

exports[`Testing Cluster Component > Snapshot > position が left の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-left_b5f122"
  >
    Cluster
  </ul>
</div>
`;

exports[`Testing Cluster Component > Snapshot > position が right の場合 1`] = `
<div>
  <ul
    class="_cluster_b5f122 _cluster-right_b5f122"
  >
    Cluster
  </ul>
</div>
`;
